.pre-commit-config.yaml

```diff
@@ -0,0 +1,21 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.5.0
+    hooks:
+    -   id: no-commit-to-branch
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-ast
+    -   id: check-case-conflict
+    -   id: check-merge-conflict
+    -   id: check-symlinks
+    -   id: check-yaml
+        exclude: '^(conda-recipe/meta.yaml)$'
```

Comment:
It might make sense to also exclude `CAproto.html` (and revert the changes to it) as it is a vendored file that we use for code-generation.

Reply:
Thoughts on this, @klauer? If you feel that we _should_ reformat `CAproto.html` I'm OK with it, but my sense it that we should omit it.

Once that's settled one way or the other, I am in favor of merging this.

Reply:
I'm 50/50 - trailing whitespace is annoying, but changing vendored files is probably worse.

I'll exclude it from that check, rebase, and force push shortly.

Reply:
Resolved in force-push (--force-with-lease, I'm not a complete monster)